bbhash_iter_test.go

```diff
@@ -0,0 +1,150 @@
+package bbhash_test
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/relab/bbhash"
+	"github.com/relab/bbhash/internal/fast"
+)
+
+func TestChunks(t *testing.T) {
+	input := "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum."
+	// Read the string and get the resulting bytes from the Chunks() method
+	r := strings.NewReader(input)
+	bufSz := 128
+	wantChunks := slices.Collect(slices.Chunk([]byte(input), bufSz))
+
+	i := 0
+	for got := range bbhash.ReadChunks(r, bufSz) {
+		// Remove empty slots to make the "got" and "want" chunks to be the same,
+		//  in case the whole chunk of bytes is not used.
+		got = bytes.Trim(got, "\x00")
+		if diff := cmp.Diff(got, wantChunks[i]); diff != "" {
+			t.Errorf("Chunks() (-got +want)\n%s", diff)
+		}
+		i++
+	}
+}
+
+func TestHashKeysFromChunksSingleWord(t *testing.T) {
+	word := "Hello"
+	hashFunc := 0 // 0 = Fasthash, 1 = sha256
+	r := strings.NewReader(word)
+	chunks := bbhash.ReadChunks(r, 8)
+
+	gotHashedKeys := bbhash.Keys(hashFunc, chunks)
+
+	// As we are dealing with an append on wordByte, som bytes are removed...
+	// ...from the initial creation such that wordByte and chunks have the same length
+	wordByte := make([]byte, 3)
+	wordByte = append([]byte(word), wordByte...)
+	wantHashedKeys := []uint64{fast.Hash64(1, wordByte)}
+
+	for i := range wantHashedKeys {
+		if diff := cmp.Diff(gotHashedKeys[i], wantHashedKeys[i]); diff != "" {
+			t.Errorf("Keys(): (-got +want) \n%s", diff)
+		}
+	}
+}
+
+func TestHashKeysFromChunksSingleWordSHA256(t *testing.T) {
+	word := "Hello"
+	hashFunc := 1 // 0 = Fasthash, 1 = sha256
+	r := strings.NewReader(word)
+	chunks := bbhash.ReadChunks(r, 8)
+
+	gotHashedKeys := bbhash.Keys(hashFunc, chunks)
+
+	// As we are dealing with an append on wordByte, som bytes are removed...
```

Comment:
This was confusing. Are you adding 3 bytes because the word ("Hello") is only 5 bytes long, in order to fill the chunk, which is 8? If so, I think there are better ways to do it. Need to think about it. The question is whether or not to return a chunk of 8 bytes when the last chunk contains only 5 bytes. I think the right thing to do is to return only 5 bytes for the last chunk and not fill the buffer. That is, I think this might do it:
```go
n, err := readerInfo.Read(buffer)
...
if !yield(buffer[:n]) {
```